NJW/DAY14/오픈채팅방.js

```diff
@@ -0,0 +1,30 @@
+function solution(record) {
+    let answer = [];
+    let user_db = {};
+
+    for (const r of record) {
+        const word = r.split(" ");
+        const cmd = word[0];
+        const uid = word[1];
+        const nickname = word[2];
+        if (cmd === "Enter" || cmd === "Change") {
+            user_db[uid] = nickname;
+        }
+    }
+
+    for (const r of record) {
+        const word = r.split(" ");
+        const cmd = word[0];
+        const uid = word[1];
+
+        const nickname = user_db[uid];
+
+        if (cmd === "Enter") {
+            answer.push(`${nickname}님이 들어왔습니다.`);
+        } else if (cmd === "Leave") {
+            answer.push(`${nickname}님이 나갔습니다.`);
+        }
+    }
```

Comment:
```suggestion
    for (const r of record) {
        const [state, id, nickname] = r.split(' ');

        updateNickname(user, state, id, nickname);
        addLog(result, state, id);
    }
```
코드를 직관적이게 잘 작성하셨네요!
다만 현재 모든 로직이 한 함수 안에 있고 record를 for문으로 두 번 순회하고 있어서
한 번 순회하면서 닉네임 업데이트와 로그 기록을 동시에 처리하면 더 효율적이고 가독성도 좋아질 것 같습니다.